compiler/rustc_trait_selection/src/error_reporting/infer/need_type_info.rs

```diff
@@ -557,12 +558,114 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                         _ => "",
                     };
 
-                    multi_suggestions.push(SourceKindMultiSuggestion::new_fully_qualified(
-                        receiver.span,
-                        def_path,
-                        adjustment,
-                        successor,
-                    ));
+                    let mut paths = vec![];
+                    let param_env = ty::ParamEnv::reveal_all();
```

Comment:
Don't use this param-env, since it will cause ICEs with const generics. Pass a param-env from the caller.